File: library/src/main/java/leavesc/hello/basis/view/IBaseViewAction.java
package leavesc.hello.basis.view;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.StringRes;
import android.widget.Toast;

/**
 * 作者：leavesC
 * 时间：2019/4/23 9:27
 * 描述：
 */
public interface IBaseViewAction {

    Context getContext();

    void showLoadingDialog(String message, boolean cancelable);

    default void showLoadingDialog(String message) {
        showLoadingDialog(message, false);
    }

    void dismissLoadingDialog();

    default <T extends Activity> void startActivity(Class<T> clazz) {
        getContext().startActivity(new Intent(getContext(), clazz));
    }

    default void showToast(String message) {
        Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
    }

    default void showToast(@StringRes int id) {
        showToast(getContext().getString(id));
    }

}

File: library/src/main/java/leavesc/hello/basis/view/IBaseViewAction.kt
package leavesc.hello.basis.view

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.widget.Toast
import androidx.annotation.StringRes

/**
 * 作者：leavesC
 * 时间：2019/6/21 10:48
 * 描述：
 */
interface IBaseViewAction {

    val context: Context

    fun showLoadingDialog(message: String, cancelable: Boolean)

    fun showLoadingDialog(message: String) {
        showLoadingDialog(message, false)
    }

    fun dismissLoadingDialog()

    fun <T : Activity> startActivity(clazz: Class<T>) {
        context.startActivity(Intent(context, clazz))
    }

    fun showToast(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    fun showToast(@StringRes id: Int) {
        showToast(context.getString(id))
    }

}
